apps/system-e2e/src/tests/islandis/application-system/acceptance/parental-leave.spec.ts

```diff
@@ -152,199 +235,479 @@ test.describe('Parental leave', () => {
+      }),
+    ).toBeVisible()
+    await page.getByTestId('employer-email').type(employer.email)
+    await page.getByTestId('employer-phone-number').type('6555555')
+    const employmentRatio = page.getByTestId('employment-ratio')
+    await employmentRatio.type('100%')
+    await employmentRatio.press('Enter')
+    await proceed()
+
+    // Who is your employer?
+    await expect(
+      page.getByRole('heading', {
+        name: label(parentalLeaveFormMessages.employer.title),
+      }),
+    ).toBeVisible()
+    await proceed()
+
+    // Additional documentation for application
+    await expect(
+      page.getByRole('heading', {
+        name: label(parentalLeaveFormMessages.attachmentScreen.title),
+      }),
+    ).toBeVisible()
+    await proceed()
+
+    // Now it is time to select the parental leave periods
+    await expect(
+      page.getByRole('region', {
+        name: label(parentalLeaveFormMessages.shared.periodsImageTitle),
+      }),
+    ).toBeVisible()
+    await proceed()
+
+    // Start of parental leave
+    await page
+      .getByRole('region', {
+        name: label(parentalLeaveFormMessages.firstPeriodStart.title),
+      })
+      .getByRole('radio', {
+        name: label(
+          parentalLeaveFormMessages.firstPeriodStart.dateOfBirthOption,
+        ),
+      })
+      .click()
+    await proceed()
+
+    // Leave duration
+    await expect(
+      page.getByRole('heading', {
+        name: label(parentalLeaveFormMessages.duration.title),
+      }),
+    ).toBeVisible()
+    await page
+      .getByRole('radio', {
+        name: label(parentalLeaveFormMessages.duration.monthsOption),
+      })
+      .click()
+    await proceed()
+
+    await expect(
+      page.getByRole('region', {
+        name: label(parentalLeaveFormMessages.duration.title),
+      }),
+    ).toBeVisible()
+    await page
+      .getByRole('button', {
+        name: '3 months',
+        exact: true,
+      })
+      .click()
+    await proceed()
+
+    // What percent off your employment ratio will you take for the leave?
+    const selectPercentageUse = page.getByTestId('select-percentage-use')
+    await selectPercentageUse.focus()
+    await page.keyboard.type('50%')
+    await page.keyboard.press('Enter')
+    await proceed()
+
+    // Here is your current leave plan
+    await expect(
+      page.getByRole('button', {
+        name: label(parentalLeaveFormMessages.leavePlan.addAnother),
+      }),
+    ).toBeVisible()
+    await proceed()
+
+    // Submit application
+    await page
+      .getByRole('button', {
+        name: label(parentalLeaveFormMessages.confirmation.title),
+      })
+      .click()
+
+    await page.waitForResponse(`${apiUrl}/api/graphql?op=SubmitApplication`)
+    const applicationSubmitSuccess = page.getByRole('heading', {
+      name: label(parentalLeaveFormMessages.finalScreen.title),
+    })
+    if (await applicationSubmitSuccess.isVisible()) {
+      await getEmployerEmailAndApprove(employer, page)
```

Comment:
Using `expect` is standard, and will automatically wait for the element to appear with sane default timeout timing.
```suggestion
    await expect(applicationSubmitSuccess).toBeVisible()
```